### PR TITLE
Update 403 error for openapi

### PR DESCRIPTION
**Pull request by astrozzc**
When request with insufficent permission is made, 403 error is returned.
Which is not described in the openapi.json. This PR is to fix this.

### CODE REVIEW

**Comment by astrozzc** [comment]
Example response: 
{
    "errors": [
        {
            "detail": "Query parameters [application] are required.",
            "source": "detail",
            "status": 400
        }
    ]
}

File: docs/source/specs/openapi.json [hunk 1]
```diff
@@ -110,6 +110,16 @@
           "401": {
             "description": "Unauthorized"
           },
+          "403": {
+            "description": "Insufficent permissions to list principals",
```
Comment (coderbydesign) [inline docs/source/specs/openapi.json]: Looks like the spelling of "Insufficent" to "Insufficient" could be updated throughout.

**Comment by astrozzc** [comment]
> Insufficient

Ah, I changed it from insufficient to insufficent 🤦‍♂️

File: docs/source/specs/openapi.json [hunk 2]
```diff
@@ -259,6 +289,16 @@
           "401": {
             "description": "Unauthorized"
           },
+          "403": {
+            "description": "Insufficent permissions to delete group",
```
[the remark this answers is not in the document]
Comment (astrozzc) [reply docs/source/specs/openapi.json]: Good catch!

File: docs/source/specs/openapi.json [hunk 3]
```diff
@@ -453,6 +493,16 @@
           "401": {
             "description": "Unauthorized"
           },
+          "403": {
+            "description": "Insufficent permissions to update principal in group",
```
[the remark this answers is not in the document]
Comment (astrozzc) [reply docs/source/specs/openapi.json]: Fixed 👍

File: docs/source/specs/openapi.json [hunk 4]
```diff
@@ -213,6 +233,16 @@
           "401": {
             "description": "Unauthorized"
           },
+          "403": {
+            "description": "Insufficent permissions to get group",
+            "content": {
+              "*/*": {
```
[the remark this answers is not in the document]
Comment (astrozzc) [reply docs/source/specs/openapi.json]: Right, thanks, this is a place I missed.

File: docs/source/specs/openapi.json [hunk 5]
```diff
@@ -213,6 +233,16 @@
           "401": {
             "description": "Unauthorized"
           },
+          "403": {
+            "description": "Insufficent permissions to get group",
```
[the remark this answers is not in the document]
Comment (astrozzc) [reply docs/source/specs/openapi.json]: Right, I missed this one when I change get to list 👍

**Pull request merged**